### DISCUSSION TITLE
Type for a page's `request`?

### DISCUSSION THREAD

**Discussion by john-prutton** [vercel/next.js#53994] [App Router] [answered]
When I hit `/pay?beerID=1` in my app and console log the `request` for the `Pay(request: Request)`, I see:
```
{
  params: {},
  searchParams: {
    beerID: "1"
  }
}
```

Although this is exactly what I want (I need searchParams for auth callback url), this does not match the shape of `Request`, nor does it match `NextRequest`.

What is the correct type of the request sent to pages in the app directory?

**Comment by icyJoseph** [comment, marked as answer]
AFAIK there's no way to access that, just now. You can access cookies and headers through special functions though.

- https://nextjs.org/docs/app/api-reference/functions/cookies
- https://nextjs.org/docs/app/api-reference/functions/headers

One thing to consider is that for pre-rendered pages, the concept of a "request" is not really applicable. Since these are pre-rendered at build time, without any incoming request whatsoever.

**Reply by john-prutton** [reply]
Thanks for the help. I realise I should be using dynamic routes for my use case.

**Comment by john-prutton** [comment]
I see that this type comes from [dynamic segments](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes#example).
